torch/_prims/executor.py

```diff
@@ -30,48 +40,51 @@ def execute(gm: GraphModule, *args, executor: str = "aten"):
 
         # PROTOTYPE nvfuser executor
         # Everything in the graph must support nvfuser
-
-        fusion = Fusion()
-        with FusionDefinition(fusion) as fd:
-
-            def _to_nvfuser_constant(arg):
-                if isinstance(arg, Number):
-                    return fd.define_constant(arg)
-                else:
-                    return arg
-
-            class FusionInterpreter(torch.fx.Interpreter):
-                def call_function(self, target, args, kwargs):
-                    args = tuple(map(_to_nvfuser_constant, args))
-                    target = target.impl_nvfuser
-                    args = (fd,) + args
-                    return target(*args, **kwargs)
-
-            def to_nv(arg):
-                if isinstance(arg, torch.Tensor):
-                    x = fd.define_tensor(
-                        arg.size(), arg.stride(), getnvFuserDtype(arg.dtype)
-                    )
-                    fd.add_input(x)
-                    return x
-                else:
-                    return arg
-
-            # Transforms graph to call nvfuser lowerings
-            # Note, this doesn't handle nested structures in the args, TODO: add tree_flatten
-            nv_args = tree_map(to_nv, args)
-            out = FusionInterpreter(gm).run(*nv_args)
-            flat_out, unflatten_spec = tree_flatten(out)
-            for o in flat_out:
-                fd.add_output(o)
-            assert len(args) == 1
-            args = args[0]  # we are passing a packed list of args
-            return tree_unflatten(
-                fusion.execute(
-                    tuple(arg for arg in args if isinstance(arg, torch.Tensor))
-                ),
-                unflatten_spec,
-            )
+        if gm in execute.fusion_cache:
```

Comment:
IIUC, we agreed to swap the nvfuser executor with the on in Ivan's PR? #80525

Reply:
yes, just signed off Ivan's PR. Will merge to use his cache.